curl/inc/leatherman/curl/client.hpp

```diff
@@ -153,6 +206,13 @@ namespace leatherman { namespace curl {
         response put(request const& req);
 
         /**
+         * Downloads the file from the specified url.
+         * @param req The HTTP request to perform 
+         * @param file_path The file that the downloaded contents will be written to. 
+         */
+        void download_file(request const& req, std::string const& file_path);
```

Comment:
Since we're writing a file, we should probably allow specifying the permissions as well. I can handle that in https://github.com/puppetlabs/leatherman/pull/249 as well if you'd prefer.